org.gecko.emf.exporter/src/org/gecko/emf/exporter/EMFExportOptions.java

```diff
@@ -21,23 +21,16 @@ public interface EMFExportOptions {
 	// locale to use
 	String OPTION_LOCALE = "LOCALE";
 
-	// export non-containment references, in addition to containment references (exported by default)
+	// export non-containment references, in addition to containment references
+	// (exported by default)
 	String OPTION_EXPORT_NONCONTAINMENT = "EXPORT_NONCONTAINMENT";
```

Comment:
In order to avoid accidentally Exporting a whole DB, the default for this should be false. 

If it is false, I further more would change the definition of NonContainment export a little bit:

If this is false, the following must apply:

```
Family simpsonFamily = createSimpsonFamily(basicFactory);
xmiResource.getContents().add(simpsonFamily);
exporterService.exportResourceTo(xmiResource, fileOutputStream, 
				Map.of(
						EMFExportOptions.OPTION_LOCALE, Locale.GERMANY						
					)
				);
```
Output: only the Family is exported, but no Persons referenced inside it.

```
Family simpsonFamily = createSimpsonFamily(basicFactory);
xmiResource.getContents().add(simpsonFamily);
xmiResource.getContents().add(simpsonFamily.getFather());
xmiResource.getContents().add(simpsonFamily.getMother());
xmiResource.getContents().addAll(simpsonFamily.getChildren());
exporterService.exportResourceTo(xmiResource, fileOutputStream, 
				Map.of(
						EMFExportOptions.OPTION_LOCALE, Locale.GERMANY						
					)
				);
```
Output: Family and The Persons will be exported, as they are part of the Resource to export.

Reply:
Non-containment is defined on EMF model itself, i.e.: 

```
<eStructuralFeatures xsi:type="ecore:EReference" (...) containment="true" (...)>
```
			
 .. which then allows to check if given EReference is a containement via: `org.eclipse.emf.ecore.EReference.isContainment()`
 		 		
That is how it was understood so far, exactly as per what EMF's definition of containment.

If we attempt to change such, shouldn't we simply name it in a different way, so at not cause confusion / clash? also, how would we check if we'd not be able to use EMF's built-in way (i.e. `org.eclipse.emf.ecore.EReference.isContainment()`) ?

Reply:
I have the feeling you are mixing things up here.

the `father, `mother` and `children` references are Non-Containments meaning:

```
<eStructuralFeatures xsi:type="ecore:EReference" (...) containment="false" (...)>
```

Thus `org.eclipse.emf.ecore.EReference.isContainment()` will always return `false`.  

So, if the `EXPORT_NONCONTAINMENT` option is not set, you need to call `get(<feature>, false)`, which means don't resolve the Object. If the return Object is no Proxy it should only be exported if it is part of the Resources content list. If it is a Proxy, you need to look if the URI matches the Resources you are exporting. If this is the case, you can use the Object you would get from `Resource.getEObject(<URI Fragment))`.

Reply:
I summarized my understanding of how 'containment' is used / understood in EMF and how it's used in the EMF Exporters' code currently. 

I was referring to example you provided, i.e.: 

> 	(...) If this is false, the following must apply: (...)
> 
> 	Family simpsonFamily = createSimpsonFamily(basicFactory);
> 	xmiResource.getContents().add(simpsonFamily);
> 	xmiResource.getContents().add(simpsonFamily.getFather());
> 	xmiResource.getContents().add(simpsonFamily.getMother());
> 	xmiResource.getContents().addAll(simpsonFamily.getChildren());
> 	exporterService.exportResourceTo(xmiResource, fileOutputStream, 
> 					Map.of(
> 							EMFExportOptions.OPTION_LOCALE, Locale.GERMANY						
> 						)
> 					);
> 
> 	Output: Family and The Persons will be exported, as they are part of the Resource to export.
> 
> 	(...)


In the `org.gecko.emf.osgi.example.model.basic` model, `Family` EClass defines 3 references (`father`, `mother` and `children`) which are marked in that EMF model as non-contained, i.e. they exist separately from `Family`. 

However, the example you provided (quoted above) I understood as: even though those 3 references are non-containment, if we add them to Resource, they are still exported, even though `EXPORT_NONCONTAINMENT` option is set to false (vide (...) _If this is false, the following must apply:_ (...)). 

Perhaps `Resource` has a different understanding of what containement means, and you're referring to that implicitly - I am referring to how that is understood in EMF model and code written so far. 

I need to know exactly what you mean.